bolero/representation/dmp_behavior.py

```diff
@@ -346,10 +346,12 @@ def imitate(self, X, Xd=None, Xdd=None, alpha=0.0,
             warnings.warn("Xdd is deprecated")
 
         X = X[:, :, 0].T.copy()
-        dmp.imitate(np.arange(0, self.execution_time + self.dt, self.dt),
-                    X, self.weights, self.widths, self.centers,
-                    alpha, self.alpha_y, self.beta_y, self.alpha_z,
-                    allow_final_velocity)
+        dmp.imitate(
+            np.linspace(0, self.execution_time + self.dt,
```

Comment:
is `self.execution_time + self.dt` what you want here?

Reply:
You're right!